### ISSUE TITLE
Add non threaded SetKeyBitmap

### ISSUE THREAD

**Issue by zegreatclan** [OpenStreamDeck/StreamDeckSharp#4]
Hi !
First thanks a lot for your lib, the stream deck would have been out interest without it :D

I've been using your library but I had some troubles
When settings the keys pictures from a dispatcher timer lot of transmission errors occurred on the Stream deck (pictures with errors, pictures on the wrong key). Adding a delay between each call seems to stabilize it, however it was slow :D

I solved it by using a custom synchronous method : 

It has two advantages, I can control the actual send rate, so instead of enqueing without knowing what I really send to the deck, I can control the "flow"outside of the Steamdeck controller, 

```c#
        public void SetKeyBitmapSync(int keyId, StreamDeckKeyBitmap bitmap)
        {
            VerifyNotDisposed();

            var page1 = StreamDeckCom.GeneratePage1(keyId, bitmap.rawBitmapData);
            var page2 = StreamDeckCom.GeneratePage2(keyId, bitmap.rawBitmapData);

            device.Write(page1);
            device.Write(page2);
        }
```

Also it would be nice to implement some kind of comparers on the StreamDeckKeyBitmap, maybe by using a dumb md5 hash of the rawBitmapData, in this way it allows the user to easily keep track of previous key states and avoid sending them again if it doesn't changes, 
I solved it like this, then I use the Hash to compare two StreamDeckKeyBitmap :

```c#
public partial class StreamDeckKeyBitmap
    {
        /// <summary>
        /// Solid black bitmap
        /// </summary>
        /// <remarks>
        /// If you need a black bitmap (for example to clear keys) use this property for better performance (in theory ^^)
        /// </remarks>
        public static StreamDeckKeyBitmap Black { get { return black; } }

        public string Hash { get; }

        private static readonly StreamDeckKeyBitmap black = new StreamDeckKeyBitmap(null);

        /// <remarks>
        /// The raw pixel format is a byte array of length 15552. This number is based on the image
        /// dimensions used by StreamDeck 72x72 pixels and 3 channels (RGB) for each pixel. 72 x 72 x 3 = 15552.
        ///
        /// The channels are in the order BGR and the pixel rows (stride) are in reverse order.
        /// If you need some help try <see cref="StreamDeckKeyBitmap"/>
        /// </remarks>
        internal readonly byte[] rawBitmapData;

        internal StreamDeckKeyBitmap(byte[] bitmapData)
        {
            if (bitmapData != null)
            {
                if (bitmapData.Length != StreamDeckHID.rawBitmapDataLength) throw new NotSupportedException("Unsupported bitmap array length");
                this.rawBitmapData = bitmapData;

                this.Hash = GetMd5Hash(this.rawBitmapData);
            }
        }

        private static string GetMd5Hash(byte[] input)
        {
            using (MD5 md5Hash = MD5.Create())
            {
                byte[] data = md5Hash.ComputeHash(input);
                StringBuilder sBuilder = new StringBuilder();
                for (int i = 0; i < data.Length; i++)
                {
                    sBuilder.Append(data[i].ToString("x2"));
                }
                return sBuilder.ToString();
            }
        }
    }
}
```

**Comment by zegreatclan** [comment]
I've just seen in details your "media player" video and the "glitches" you had were exactly what I was facing. I thought about a mis-use from my code, but it looks like messages are mixing in a way due to the tasks.

**Comment by wischi-chr** [comment]
I would love to make it syncron but it's to slow for my taste. Try the video demo (from the wiki) with a syncronous method. It's pretty slow at least in my last tests it was (that was before I optimized hidlib a bit).

Will do some tests tomorrow ;-)

**Comment by zegreatclan** [comment]
Concerning synchronous, indeed, I just made a test when sending all the buttons everytime and ... well it is slow, but never glitches.  I will try to look if i can solve the glitch problem in the threaded way, Since messages are sent in two packets I guess they are mixing up and not received in the right order by the streamdeck

Concerning the hash, I agree I would not filter them on the library side, if user wan't to send it, it has to be sent :D,about performances maybe lazy loading the hash when requesting it could be a nice compromise, today since the byte array is internal it's impossible to do this comparison work (when it comes from a bitmap it's easy to do it on the bitmap itself, but when it comes to wpf (which works very nicely by the way !), it's impossible to guess it without having access to the array, a comparer or anything allowing to compare them.

**Comment by wischi-chr** [comment]
Regarding image glitches, I don't think that this happens because they are getting mixed up. I think the problem is, that the previous operation is interruped because new data is already comming in and that rips apart the bitmap. I'm doing some tests now. Maybe I find a solution ;-)

About the hashes. I added `Equals`, `==` and a custom `GetHashCode` to compare bitmap objects by content _(see last commits - not released on nuget yet)_

Please let me know it that would help you.

**Comment by zegreatclan** [comment]
Amazing ! Thanks a lot ! I will test it when I get back from work. It couldn't be better, iequatable, operators , lazy loading of the hash, just perfect !

**Comment by wischi-chr** [comment]
I just released v0.1.9 to fix the image glitches.

Details about the problem in the wiki:  
https://github.com/OpenStreamDeck/StreamDeckSharp/wiki/Maximum-update-rate

Pls let me know if that works for you as well. Thx.
